### PR TITLE
139 - Implement Cart Api

### PR DESCRIPTION
**Pull request by kristher1619**
- Cart Model Api
- 139 Add Cart Api Controller

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -3,10 +3,12 @@
...
 /vendor
+/tests/coverage
 .env
 .env.backup
 .phpunit.result.cache
 Homestead.json
 Homestead.yaml
 npm-debug.log
 yarn-error.log
+coverage.xml
```
Comment (jerud3v) [inline .gitignore:14]: please add `.idea/`

File: app/Models/Cart.php [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+        $cart = self::exist($userId, $product['product_id'])->first();
+        if ($cart) {
+            $cart->increment('quantity', $quantity);
+            return $cart;
+        }
+
+        // do nothing if $quantity is less than 1
+        if ($quantity < 1) {
+            return null;
+        }
```
Comment (jerud3v) [inline app/Models/Cart.php:26]: Is this acceptable?
`$quantity < 1 ? null : $quantity;`
or
`$quantity >= 1 ?: null;` as thought by Sir @artfred16

File: tests/Feature/controllers/CartControllerTest.php [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+    /** @test */
+    public function it_should_remove_item_from_cart()
+    {
+        $this->post('/api/cart/', $this->product + ['quantity' => 1])->assertStatus(Response::HTTP_ACCEPTED);
+        $this->delete('/api/cart/1', ['product_id' => $this->product['product_id']])->assertStatus(Response::HTTP_ACCEPTED);
+        $this->assertDatabaseCount('carts', 0);
+    }
+
+    /** @test */
+    public function it_should_remove_mulwtiple_items_from_cart()
```
Comment (jerud3v) [inline tests/Feature/controllers/CartControllerTest.php]: typo _mulwtiple_

**Review by jerud3v — APPROVED** [review]

File: app/Models/Cart.php (hunk 2 above)
Comment (kristher1619) [reply app/Models/Cart.php:26]: nope in this case we should do nothing if $quantity is less than 1

**Pull request merged**